File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        mini = prices[0]
        maxDiff = 0
        for i in range(1,len(prices)):
            maxDiff = max(prices[i]-mini,maxDiff)
            if prices[i] < mini:
                mini = prices[i]
                
        return maxDiff

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minimum = prices[0];
        int ans = 0;
        for ( int i=1; i<prices.length; i++){
            ans = Math.max(ans, prices[i] - minimum);
            minimum = Math.min(minimum, prices[i]);
        }

        return ans;
        
    }
}